avm/ptn/azd/ml-ai-environment/modules/hub.bicep

```
@@ -0,0 +1,143 @@
+@description('Required. The AI Studio Hub Resource name.')
+param name string
+
+@description('Optional. The display name of the AI Studio Hub Resource.')
+param displayName string = name
+
+@description('Required. The storage account ID to use for the AI Studio Hub Resource.')
+param storageAccountResourceId string
+
+@description('Required. The key vault ID to use for the AI Studio Hub Resource.')
+param keyVaultResourceId string
+
+@description('Optional. The application insights ID to use for the AI Studio Hub Resource.')
+param applicationInsightsResourceId string = ''
+
+@description('Optional. The container registry ID to use for the AI Studio Hub Resource.')
+param containerRegistryResourceId string = ''
+
+@description('Required. The OpenAI Cognitive Services account name to use for the AI Studio Hub Resource.')
+param openAiName string
+
+@description('Optional. The Azure Cognitive Search service name to use for the AI Studio Hub Resource.')
+param aiSearchName string = ''
+
+@description('Required. The Azure Cognitive Search service connection name to use for the AI Studio Hub Resource.')
+param aiSearchConnectionName string
+
+@description('Required. The OpenAI Content Safety connection name to use for the AI Studio Hub Resource.')
+param openAiContentSafetyConnectionName string
+
+@description('Optional. The SKU name to use for the AI Studio Hub Resource.')
+param skuName string = 'Basic'
+
+@description('Optional. The SKU tier to use for the AI Studio Hub Resource.')
+@allowed(['Basic', 'Free', 'Premium', 'Standard'])
+param skuTier string = 'Basic'
+
+@description('Optional. The public network access setting to use for the AI Studio Hub Resource.')
+@allowed(['Enabled', 'Disabled'])
+param publicNetworkAccess string = 'Enabled'
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+// ================ //
+// Resources        //
+// ================ //
+
+resource openAi 'Microsoft.CognitiveServices/accounts@2023-05-01' existing = {
+  name: openAiName
+}
+
+resource search 'Microsoft.Search/searchServices@2021-04-01-preview' existing = if (!empty(aiSearchName)) {
+  name: aiSearchName
+}
+
+resource hub 'Microsoft.MachineLearningServices/workspaces@2024-04-01' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: skuName
+    tier: skuTier
+  }
+  kind: 'Hub'
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {
+    friendlyName: displayName
+    storageAccount: storageAccountResourceId
+    keyVault: keyVaultResourceId
+    applicationInsights: !empty(applicationInsightsResourceId) ? applicationInsightsResourceId : null
+    containerRegistry: !empty(containerRegistryResourceId) ? containerRegistryResourceId : null
```

I'd suggest to make the 2 parameters nullable `?` which would allow you to simplify the code as shown below:

```suggestion
    applicationInsights: applicationInsightsResourceId
    containerRegistry: containerRegistryResourceId
```

**Only** commit if you also change the parameters.

---

The test for minimizing parameters is that the values ​​of `applicationInsightsResourceId` and `containerRegistryResourceId` set in the main.bicep file are both empty strings `' '`. If do not add `!empty( ) ?` judgment, an error will be reported: `"Property id '' at path 'properties.containerRegistry' is invalid.`

Add this `!empty() ?` condition in case the parameter is an empty string `' '`.

---

Gotcha, so then you'd need to add the check where you pass the value in. Thanks for elaborating 💪, will resolve